ark/utils/segmentation_utils.py

```diff
@@ -193,41 +192,30 @@ def visualize_segmentation(segmentation_labels_xr, channel_data_xr,
     Args:
         segmentation_labels_xr (xarray.DataArray):
             xarray containing segmentation labels
-        channel_data_xr (xarray.DataArray):
-            xarray containing TIFs
         output_dir (str):
             path to directory where the output will be saved
-        chan_list (list):
-            list of channels to overlay segmentation output over
         fovs (list):
             list of FOVs to subset in segmentation_labels_xr
-        show (bool):
-            whether or not to show plot
     """
 
+    # assign fovs and channels to everything if None
     if fovs is None:
         fovs = segmentation_labels_xr.fovs.values
-    misc_utils.verify_in_list(fovs=fovs, segmentation_label_fovs=segmentation_labels_xr.fovs)
 
-    if chan_list is not None:
-        misc_utils.verify_in_list(channels=chan_list,
-                                  channel_data_channels=channel_data_xr.channels)
+    # verify that fovs and channels exist in their respective xarrays
+    misc_utils.verify_in_list(fovs=fovs, segmentation_label_fovs=segmentation_labels_xr.fovs)
 
     for fov in fovs:
+        # generates segmentation borders and labels
         labels = segmentation_labels_xr.loc[fov, :, :, 'whole_cell'].values
-        # If chan_list is provided, overlay segmentation output over it
-        if chan_list is not None:
-            input_data = channel_data_xr.loc[fov, :, :, chan_list].values
-            save_path = '_'.join([f'{fov}', *chan_list.astype('str'), 'overlay.tiff'])
-            plot_utils.plot_overlay(
-                labels,
-                plotting_tif=input_data,
-                path=os.path.join(output_dir, save_path), show=show
-            )
-        # Generates segmentation borders and labels
-        plot_utils.plot_overlay(
+
+        # save the labels respectively
+        io.imsave(os.path.join(output_dir, f'{fov}_segmentation_labels.tiff'), labels)
+
+        # generate segmentation borders and labels
+        contour_mask = plot_utils.create_overlay(
```

Comment:
We can pull this functionality out directly, rater than using create_overlay to do it, since it's just a call to skimage.find_boundaries

Reply:
A minor problem here is that `create_overlay` needs the result of `find_boundaries` to figure out which values in `rescaled` to set to white. Two ways around this:

1. Live with duplicated code in `create_overlay`
2. Add an argument `predicted_contour_mask` to `create_overlay` so we can pass in the result of `find_boundaries` in `save_segmentation_labels` in (however, the `find_boundaries` logic now needs to be duplicated in the notebook call to `create_overlay`)

Reply:
Yea, it's fine to call `find_boundaries` twice, it's very fast

Reply:
I'd just have both functions calculate it